### PR TITLE
Add org-slug and org-id flags to orb validate & process commands

### PR DESCRIPTION
**Pull request by zbenhadi**
# Checklist

=========

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have checked for similar issues and haven't found anything relevant.
- [x] This is not a security issue (which should be reported here: https://circleci.com/security/)
- [x] I have read [Contribution Guidelines](https://github.com/CircleCI-Public/circleci-cli/blob/main/CONTRIBUTING.md).

### Internal Checklist
- [x] I am requesting a review from my own team as well as the owning team
- [x] I have a plan in place for the monitoring of the changes that I am making (this can include new monitors, logs to be aware of, etc...)

## Changes

=======

- Added a new orbOrgOptions struct, to represent org-slug and org-id values
- Added a new api package: collaborators - presents an interface with a single method "GetOrgCollaborations"
- Changed "ValidateOrb" GQLQuery to include "ownerId" param, based on org-slug and org-id flags

## Rationale

=========

Integrate the capability to validate an orb when using other private orbs from the same namespace.

### CODE REVIEW

File: cmd/orb.go [hunk 1]
```diff
@@ -1746,3 +1792,29 @@ func orbInformThatOrbCannotBeDeletedMessage() {
...
 }
+
+func (o *orbOptions) getOrgId(orgInfo orbOrgOptions) (string, error) {
+	if orgInfo.OrgID == "" && orgInfo.OrgSlug == "" {
+		return "", nil
+	}
+
+	var orgID string
+	if strings.TrimSpace(orgInfo.OrgID) != "" {
+		orgID = orgInfo.OrgID
```
Comment (JulesFaucherre) [inline cmd/orb.go:1803]: Nitpicking: you could return directly here, avoiding too much nesting with the following `else if`

**Review by JulesFaucherre — CHANGES_REQUESTED** [review]
Capture d’écran 2023-04-26 à 14 30 17
It seems the usage ask to use the `<vcs-name>/<org-name>` but the code only supports the `<vcs-short>/<org-name>` syntax

File: cmd/orb.go (hunk 1 above)
Comment (zbenhadi) [reply cmd/orb.go]: Addressed 👍

**Review by JulesFaucherre — APPROVED** [review]
Very great job, thanks for also fixing the slug issue on config validate!

**Pull request merged**